pkg/ccl/crosscluster/logical/range_stats.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package logical
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/repstream/streampb"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+)
+
+type rangeStatsByProcessorID struct {
+	mu             syncutil.Mutex
```

Comment:
Is the mutex required here? Are rows and producer metas produced in parallel?

Reply:
i don't know if they can produced in parallel, but i thought better safe than sorry, especially because these apis aren't called too often.